src/internal/Observable.ts

```diff
@@ -347,4 +347,126 @@ export class Observable<T> implements Subscribable<T> {
   pipe(...operations: OperatorFunction<any, any>[]): Observable<any> {
     return pipeFromArray(operations)(this);
   }
+
+  /**
+   * Observable is async iterable, so it can be used in `for await` loop. This method
+   * of subscription is cancellable by breaking the for await loop. Although it's not
+   * recommended to use Observable's AsyncIterable contract outside of `for await`, if
+   * you're consuming the Observable as an AsyncIterable, and you're _not_ using `for await`,
+   * you can use the `throw` or `return` methods on the `AsyncGenerator` we return to
+   * cancel the subscription. Note that the subscription to the observable does not start
+   * until the first value is requested from the AsyncIterable.
+   *
+   * Functionally, this is equivalent to using a {@link concatMap} with an `async` function.
+   * That means that while the body of the `for await` loop is executing, any values that arrive
+   * from the observable source will be queued up, so they can be processed by the `for await`
+   * loop in order. So, like {@link concatMap} it's important to understand the speed your
+   * source emits at, and the speed of the body of your `for await` loop.
+   *
+   * ## Example
+   *
+   * ```ts
+   * import { interval } from 'rxjs';
+   *
+   * async function main() {
+   *  // Subscribe to the observable using for await.
+   *  for await (const value of interval(1000)) {
+   *    console.log(value);
+   *
+   *    if (value > 5) {
+   *      // Unsubscribe from the interval if we get a value greater than 5
+   *      break;
+   *    }
+   *  }
+   * }
+   *
+   * main();
+   * ```
+   */
+  [Symbol.asyncIterator](): AsyncGenerator<T, void, void> {
+    let subscription: Subscription | undefined;
+    let hasError = false;
+    let error: any;
```

Comment:
If errors should be typed `unknown` in V8, we might wanna do it right away:

```suggestion
    let error: unknown;
```